### ISSUE TITLE
Is the cifar10_cnn.py example actually performing data augmentation?

### ISSUE THREAD

**Issue by asmith26** [fchollet/keras#2893]
When `datagen.fit(X_train)` is called in the [`cifar10_cnn.py` example](https://github.com/fchollet/keras/blob/master/examples/cifar10_cnn.py#L103), shouldn't it be (when `data_augmentation=True`):

```python
datagen.fit(X_train, augment=True)
```

as the [default value for `augment` is `False`](https://github.com/fchollet/keras/blob/master/keras/preprocessing/image.py#L410)?

Also, I am right in thinking when using `augment=True` the original (i.e. non-augmented - ignoring any normalisation/standardisation) data is not necessarily trained on? If so, I thought data augmentation is a method of artificially increasing the size of your dataset, so shouldn't we additionally be training on the non-augmented data? Thanks

- [x] Check that you are up-to-date with the master branch of Keras. You can update with:
pip install git+git://github.com/fchollet/keras.git --upgrade --no-deps

- [x] If running on Theano, check that you are up-to-date with the master branch of Theano. You can update with:
pip install git+git://github.com/Theano/Theano.git --upgrade --no-deps

- [x] Provide a link to a GitHub Gist of a Python script that can reproduce your issue (or just copy the script here if it is short).

**Comment by joelthchao** [comment]
`fit` is only needed when any of `featurewise_center`, `samplewise_center`, `featurewise_std_normalization`, `samplewise_std_normalization`, or `zca_whitening` is set `True`. Therefore, in this example, you can even remove that line.

**Comment by asmith26** [comment]
Thanks @joelthchao - what about the arguments that are set to true, i.e. (lines96-98):

```python
width_shift_range=0.1,  # randomly shift images horizontally (fraction of total width)
height_shift_range=0.1,  # randomly shift images vertically (fraction of total height)
horizontal_flip=True,  # randomly flip images
```

To ensure the ImageDataGenerator does augment the input data according to this specification, shouldn't the example use:

```python
datagen.fit(X_train, augment=True)
```
?

**Comment by joelthchao** [comment]
Please look into code. What `fit` does is assign value to `self.mean`, `self.std` or `self.principal_components`, and none of them will be used when 
```python
width_shift_range=0.1,  # randomly shift images horizontally (fraction of total width)
height_shift_range=0.1,  # randomly shift images vertically (fraction of total height)
horizontal_flip=True,  # randomly flip images
```

**Comment by asmith26** [comment]
>What fit does is assign value to self.mean, self.std or self.principal_component

Thanks I see.

I still can't find where data augmentation takes place though:
1. Does it take place in `fit` when `augment=True` (since this makes a call to `self.random_transform()`:
```python
if augment:
            aX = np.zeros(tuple([rounds * X.shape[0]] + list(X.shape)[1:]))
            for r in range(rounds):
                for i in range(X.shape[0]):
                    aX[i + r * X.shape[0]] = self.random_transform(X[i])
            X = aX
```
2. Does it take place in the `next()` method of a `ImageDataGenerator` object? I see this [makes a call to `self.random_transform()`](https://github.com/fchollet/keras/blob/master/keras/preprocessing/image.py#L307), but I can't find where the `next()` method is called?

**Comment by joelthchao** [comment]
`random_transform` is called in [`next()`](https://github.com/fchollet/keras/blob/master/keras/preprocessing/image.py#L464), which makes `self` iterable. Please read [Iterator and Generator](http://anandology.com/python-practice-book/iterators.html#the-iteration-protocol).

**Comment by asmith26** [comment]
Ok I think I've got it now - the `augment=True` in the `fit` method is for ensuring we initialise our statistics on augmented data. Thanks for help, and sorry for wasting anyone’s time.
